docs/design/2022-11-17-ttl-table.md

```diff
@@ -0,0 +1,369 @@
+# Proposal: Support TTL Table
+- Author(s): [lcwangchao](https://github.com/lcwangchao)
+- Tracking Issue: https://github.com/pingcap/tidb/issues/39262
+
+## Table of Contents
+
+<!-- TOC -->
+* [Proposal: Support TTL Table](#proposal--support-ttl-table)
+  * [Table of Contents](#table-of-contents)
+  * [Introduction](#introduction)
+  * [Detailed Design](#detailed-design)
+    * [Syntax](#syntax)
+      * [Create TTL Table](#create-ttl-table)
+      * [Alter a Table with TTL](#alter-a-table-with-ttl)
+      * [Alter to a non-TTL Table](#alter-to-a-non-ttl-table)
+      * [Constraints](#constraints)
+    * [TTL Job Management](#ttl-job-management)
+    * [TTL Job Details](#ttl-job-details)
+      * [Scan Tasks](#scan-tasks)
+      * [Delete Tasks](#delete-tasks)
+    * [New System Variables](#new-system-variables)
+    * [New Metrics](#new-metrics)
+  * [Known Issues](#known-issues)
+  * [Future Works](#future-works)
+  * [Alternative Solutions](#alternative-solutions)
+<!-- TOC -->
+
+## Introduction
+
+The rows in a TTL table will be deleted automatically when they are expired. It is useful for some scenes, for example, delete the expired verification codes. A TTL table will have a column with the type DATE/DATETIME/TIMESTAMP which will be compared with the current time, if the interval between them exceeds some threshold, the corresponding row will be deleted.
+
+## Detailed Design
+
+### Syntax
+
+#### Create TTL Table
+
+The following example shows how to create a TTL table. The column `create_at` is used to specify the creation time of the rows which will be deleted 3 months after that.
+
+```sql
+CREATE TABLE t1 (
+    id int PRIMARY KEY,
+    created_at TIMESTAMP
+) TTL = `created_at` + INTERVAL 3 MONTH;
```

Comment:
can we pass -3 MONTH here? 

btw, seem here should be "3 MONTHS", plural.

Reply:
btw, I prefer Spanner's TTL Clause https://cloud.google.com/spanner/docs/ttl/working-with-ttl#postgresql
creating a TTL "ON" a column may be more straightforward.

Reply:
> can we pass -3 MONTH here?

It is allowed currently, but meaningless in most times. If one row is created using the real time, it will be deleted after next job scheduled, so it equals to setting the expire interval to a very small value. If user sets a time in one row with a past time, it may be some different, but I don't know any scene requires it. Maybe we can disallow user to setting it a a negative value or just leave it alone because it will not bring some big problem.

> btw, seem here should be "3 MONTHS", plural.

We are using an expression to parse the duration, so, it supports both `3 MONTH` and `3 MONTHS`. I can update it plural in doc to make it more natural

> btw, I prefer Spanner's TTL Clause https://cloud.google.com/spanner/docs/ttl/working-with-ttl#postgresql
creating a TTL "ON" a column may be more straightforward.

I do not have a strong preference, PTAL @SunRunAway

Reply:
> btw, I prefer Spanner's TTL Clause https://cloud.google.com/spanner/docs/ttl/working-with-ttl#postgresql
creating a TTL "ON" a column may be more straightforward.

I prefer keeping it as an **expression** syntax instead of a particular syntax because there may be more syntax extensions in the future if you don't have a strong opinion.